src/monodroid/jni/monodroid-glue.cc

```diff
@@ -2635,6 +2641,27 @@ MonodroidRuntime::get_java_class_name_for_TypeManager (jclass klass)
 	return ret;
 }
 
+void
+MonodroidRuntime::monodroid_javasystem_loadLibrary (const char *libname) noexcept
+{
+	JNIEnv *env = osBridge.ensure_jnienv ();
+	jmethodID loadLibraryID = env->GetStaticMethodID (mono_android_Runtime, "loadLibrary", "(Ljava/lang/String;)Z");
+	if (loadLibraryID == nullptr) {
+		log_error (LOG_DEFAULT, "Failed to look up the mono.android.Runtime.loadLibrary Java method");
+		return;
+	}
+
+	jstring libname_java = env->NewStringUTF (libname);
+	jboolean success = env->CallStaticBooleanMethod (mono_android_Runtime, loadLibraryID, libname_java);
+	if (success != 0) {
```

Comment:
We should `env->DeleteLocalRef(libname_java)` here, to prevent LREF leaks.  This function is called from Managed code, so we might not "return to Java" anytime soon (if ever), which is the only other time that JNI Local References will be freeed.

Reply:
I was wondering about that actually. We pass the string to Java in `CallStaticBooleanMethod` - won't it take ownership and release the LREF when the string is garbage collected?

Reply:
No: https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/design.html#global_and_local_references

> Local references are valid for the duration of a native method call, and are automatically freed after the native method returns.
> …
> However, there are times when the programmer should explicitly free a local reference. …
>   * A native method creates a large number of local references, although not all of them are used at the same time.

We're (kinda) in that bullet point, but differently so:

 1. Android only allows 512 JNI Local References (per thread) to exist at a time, but more importantly *for us*
 2. JNI Local References are only collected by the JVM when execution returns to Java.

(2) is potentially *rare* for us, particularly here: a `System.Threading.Thread` (POSIX Thread) running managed code which calls *into* Java will *never* "return" to Java.  Consequently, if *we* don't explicitly free them, the number of JNI Local References may constantly increase, resulting in a possible Android JNI Local Reference table overflow + app abort.